ray-operator/controllers/ray/common/association.go

```diff
@@ -159,3 +164,23 @@ func RayJobRayClusterNamespacedName(rayJob *rayv1.RayJob) types.NamespacedName {
 		Namespace: rayJob.Namespace,
 	}
 }
+
+// GetRayClusterHeadPod get a *corev1.Pod from a *rayv1.RayCluster. Note that it returns (nil, nil) in the case of no head pod exists.
+func GetRayClusterHeadPod(ctx context.Context, reader client.Reader, instance *rayv1.RayCluster) (*corev1.Pod, error) {
+	logger := ctrl.LoggerFrom(ctx)
+
+	runtimePods := corev1.PodList{}
+	filterLabels := RayClusterHeadPodsAssociationOptions(instance)
+	if err := reader.List(ctx, &runtimePods, filterLabels.ToListOptions()...); err != nil {
+		return nil, err
+	}
+	if len(runtimePods.Items) == 0 {
+		logger.Info(fmt.Sprintf("Found %d head pods. cluster name %s, filter labels %v", len(runtimePods.Items), instance.Name, filterLabels))
```

Comment:
Avoid using `fmt.Sprintf`. We can use:

```go
logger.Info("Found 0 head Pod", "filter labels", filterLabels)
```

instead. I removed `instance.Name` because I think all KubeRay logs currently include the information about the CR name.